tests/load/pipeline/test_replace_disposition.py

```diff
@@ -0,0 +1,165 @@
+import dlt, os, pytest
+from typing import Any, Dict
+
+from tests.pipeline.utils import  assert_load_info
+from tests.load.pipeline.utils import  load_table_counts
+from tests.utils import ALL_DESTINATIONS
+from tests.load.pipeline.utils import STAGING_AND_NON_STAGING_COMBINATIONS, STAGING_COMBINATION_FIELDS
+
+@pytest.mark.parametrize(STAGING_COMBINATION_FIELDS, STAGING_AND_NON_STAGING_COMBINATIONS)
+def test_replace_disposition(destination: str, staging: str, file_format: str, bucket: str, settings: Dict[str, Any]) -> None:
+
+    # only allow 40 items per file
+    os.environ['DATA_WRITER__FILE_MAX_ITEMS'] = "40"
+    # use staging tables for replace
+    os.environ['DESTINATION__REPLACE_STRATEGY'] = "staging"
+
+    # snowflake requires gcs prefix instead of gs in bucket path
+    if destination == "snowflake":
```

Comment:
maybe we should normalize paths in copy jobs in snowflake? it has a specialized one